File: Algorithm/baekjoon/1966_프린터큐.py
# baekjoon source = "https://www.acmicpc.net/problem/1966"
'''
3
1 0
5
4 2
1 2 3 4
4 2
2 1 4 3
'''
import sys

T = int(sys.stdin.readline())
for _ in range(1, T + 1):

    N, M = map(int,sys.stdin.readline().split())
    q = list(map(int,sys.stdin.readline().split()))
    for i in range(N):
        q[i] = (q[i],i)

    idx = 0
    while q:
        if q[0] == max(q,key=lambda x: x[0]):
            t = q.pop(0)
            idx += 1
            if t[1] == M:
                break
        else:
            q.append(q.pop(0))

    print(idx)

File: Java/baekjoon/화목한스터디/src/큐/_1966_프린터큐.java
//baekjoon source = "https://www.acmicpc.net/problem/1966"
package 큐;

import java.io.BufferedReader;
import java.io.BufferedWriter;
import java.io.IOException;
import java.io.InputStreamReader;
import java.io.OutputStreamWriter;
import java.util.LinkedList;
import java.util.Queue;
import java.util.StringTokenizer;

public class _1966_프린터큐 {
	public static void main(String[] args) throws IOException {
		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
		BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));

		int t = Integer.parseInt(br.readLine());
		for (int test_case = 0; test_case < t; test_case++) {
			Queue<Integer> q = new LinkedList<>();
			StringTokenizer st = new StringTokenizer(br.readLine());
			int n = Integer.parseInt(st.nextToken()), m = Integer.parseInt(st.nextToken()) + 1;
			int[] check = new int[101];
			st = new StringTokenizer(br.readLine());
			int max = Integer.MIN_VALUE;
			for (int i = 0; i < n; i++) {
				int c = Integer.parseInt(st.nextToken());
				check[c]++;
				q.add(c + (i + 1) * 1000);
				max = Math.max(max, c);
			}

			int result = 1;
			while (true) {
				int re = q.poll();
				if (re % 1000 == max) {
					if (re / 1000 == m) {
						break;
					} else {
						result++;
						if (check[max] >= 2) {
							check[max]--;
						} else {
							check[max]--;
							for (int i = 100; i > 0; i--) {
								if (check[i] != 0) {
									max = i;
									break;
								}
							}
						}
					}
				} else {
					q.add(re);
				}
			}

			bw.write(result + "\n");
			bw.flush();
		}
	}
}
